File: TOMIMSV4/Entity/Mantenimientos/Empresa_Transporte_Piloto/clsBeEmpresa_transporte_pilotos.vb
Public Class clsBeEmpresa_transporte_pilotos
    Implements ICloneable
    Public Property IdPiloto() As Integer = 0
    Public Property IdEmpresaTransporte() As Integer = 0
    Public Property Nombres() As String = ""
    Public Property Apellidos() As String = ""
    Public Property Telefono() As String = ""
    Public Property Correo_electronico() As String = ""
    Public Property No_carnet() As String = ""
    Public Property Fecha_expiracion_carnet() As Date = Date.Now
    Public Property No_dpi() As String = ""
    Public Property No_Licencia() As String = ""
    Public Property Fecha_expiracion_licencia() As Date = Date.Now
    Public Property Codigo_barra() As String = ""
    Public Property Direccion() As String = ""
    Public Property Foto() As Byte()
    Public Property Fecha_nacimiento() As Date = Date.Now
    Public Property Fecha_ingreso() As Date = Date.Now
    Public Property Fecha_salida() As Date = Date.Now
    Public Property IdTipoLicencia() As String = ""
    Public Property User_agr() As String = ""
    Public Property Fec_agr() As Date = Date.Now
    Public Property User_mod() As String = ""
    Public Property Fec_mod() As Date = Date.Now
    Public Property Activo() As Boolean = False
    Public Property IsNew As Boolean = False
    Public Property EscaneoDocumento As Boolean = False

    Sub New()
    End Sub
    Sub New(ByRef IdPiloto As Integer, ByVal IdEmpresaTransporte As Integer, ByVal nombres As String, ByVal apellidos As String, ByVal telefono As String, ByVal correo_electronico As String, ByVal no_carnet As String, ByVal fecha_expiracion_carnet As Date, ByVal no_dpi As String, ByVal no_licencia As String, ByVal fecha_expiracion_licencia As Date, ByVal codigo_barra As String, ByVal direccion As String, ByVal foto As Byte(), ByVal fecha_nacimiento As Date, ByVal fecha_ingreso As Date, ByVal fecha_salida As Date, ByVal IdTipoLicencia As String, ByVal user_agr As String, ByVal fec_agr As Date, ByVal user_mod As String, ByVal fec_mod As Date, ByVal activo As Boolean)
        Me.IdPiloto = IdPiloto
        Me.IdEmpresaTransporte = IdEmpresaTransporte
        Me.Nombres = nombres
        Me.Apellidos = apellidos
        Me.Telefono = telefono
        Me.Correo_electronico = correo_electronico
        Me.No_carnet = no_carnet
        Me.Fecha_expiracion_carnet = fecha_expiracion_carnet
        Me.No_dpi = no_dpi
        Me.No_Licencia = no_licencia
        Me.Fecha_expiracion_licencia = fecha_expiracion_licencia
        Me.Codigo_barra = codigo_barra
        Me.Direccion = direccion
        Me.Foto = foto
        Me.Fecha_nacimiento = fecha_nacimiento
        Me.Fecha_ingreso = fecha_ingreso
        Me.Fecha_salida = fecha_salida
        Me.IdTipoLicencia = IdTipoLicencia
        Me.User_agr = user_agr
        Me.Fec_agr = fec_agr
        Me.User_mod = user_mod
        Me.Fec_mod = fec_mod
        Me.Activo = activo
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function
End Class

File: WMS.EntityCore/Datos_Maestros/clsBeEmpresa_transporte_pilotos.cs
using System.ComponentModel;
using System.ComponentModel.DataAnnotations.Schema;

namespace WMSWebAPI.Be
{
    public class clsBeEmpresa_transporte_pilotos : ICloneable
    {
        [Column("IdPiloto")]
        [DisplayName("IdPiloto")]
        public int IdPiloto { get; set; } = 0;

        [Column("IdEmpresaTransporte")]
        [DisplayName("IdEmpresaTransporte")]
        public int IdEmpresaTransporte { get; set; } = 0;

        [Column("nombres")]
        [DisplayName("nombres")]
        public string Nombres { get; set; } = "";

        [Column("apellidos")]
        [DisplayName("apellidos")]
        public string Apellidos { get; set; } = "";

        [Column("telefono")]
        [DisplayName("telefono")]
        public string Telefono { get; set; } = "";

        [Column("correo_electronico")]
        [DisplayName("correo_electronico")]
        public string Correo_electronico { get; set; } = "";

        [Column("no_carnet")]
        [DisplayName("no_carnet")]
        public string No_carnet { get; set; } = "";

        [Column("fecha_expiracion_carnet")]
        [DisplayName("fecha_expiracion_carnet")]
        public DateTime Fecha_expiracion_carnet { get; set; } = DateTime.Now;

        [Column("no_dpi")]
        [DisplayName("no_dpi")]
        public string No_dpi { get; set; } = "";

        [Column("no_licencia")]
        [DisplayName("no_licencia")]
        public string No_licencia { get; set; } = "";

        [Column("fecha_expiracion_licencia")]
        [DisplayName("fecha_expiracion_licencia")]
        public DateTime Fecha_expiracion_licencia { get; set; } = DateTime.Now;

        [Column("codigo_barra")]
        [DisplayName("codigo_barra")]
        public string Codigo_barra { get; set; } = "";

        [Column("direccion")]
        [DisplayName("direccion")]
        public string Direccion { get; set; } = "";

        [Column("foto")]
        [DisplayName("foto")]
        public byte[] Foto { get; set; }= Array.Empty<byte>();

        [Column("fecha_nacimiento")]
        [DisplayName("fecha_nacimiento")]
        public DateTime Fecha_nacimiento { get; set; } = DateTime.Now;

        [Column("fecha_ingreso")]
        [DisplayName("fecha_ingreso")]
        public DateTime Fecha_ingreso { get; set; } = DateTime.Now;

        [Column("fecha_salida")]
        [DisplayName("fecha_salida")]
        public DateTime Fecha_salida { get; set; } = DateTime.Now;

        [Column("IdTipoLicencia")]
        [DisplayName("IdTipoLicencia")]
        public string IdTipoLicencia { get; set; } = "";

        [Column("user_agr")]
        [DisplayName("user_agr")]
        public string User_agr { get; set; } = "";

        [Column("fec_agr")]
        [DisplayName("fec_agr")]
        public DateTime Fec_agr { get; set; } = DateTime.Now;

        [Column("user_mod")]
        [DisplayName("user_mod")]
        public string User_mod { get; set; } = "";

        [Column("fec_mod")]
        [DisplayName("fec_mod")]
        public DateTime Fec_mod { get; set; } = DateTime.Now;

        [Column("activo")]
        [DisplayName("activo")]
        public bool Activo { get; set; } = false;
        public bool IsNew { get; set; } = true;

        public clsBeEmpresa_transporte_pilotos() { }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}
